server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java

```diff
@@ -73,23 +87,207 @@ public LoggerContextListener enableJulChangePropagation(LoggerContext loggerCont
+    rootLogger.addAppender(fileAppender);
+    return fileAppender;
+  }
+
+  public FileAppender<ILoggingEvent> newFileAppender(LoggerContext ctx, Props props, LogbackHelper.RootLoggerConfig config, String logPattern) {
+    RollingPolicy rollingPolicy = createRollingPolicy(ctx, props, config.getFileNamePrefix());
+    FileAppender<ILoggingEvent> fileAppender = rollingPolicy.createAppender("file_" + config.getFileNamePrefix());
+    fileAppender.setContext(ctx);
+    PatternLayoutEncoder fileEncoder = new PatternLayoutEncoder();
+    fileEncoder.setContext(ctx);
+    fileEncoder.setPattern(logPattern);
+    fileEncoder.start();
+    fileAppender.setEncoder(fileEncoder);
+    fileAppender.start();
+    return fileAppender;
+  }
+
+  /**
+   * Make the logback configuration for a sub process to correctly push all its logs to be read by a stream gobbler
+   * on the sub process's System.out.
+   *
+   * @see #buildLogPattern(RootLoggerConfig)
+   */
+  public void configureForSubprocessGobbler(Props props, String logPattern) {
+    if (isAllLogsToConsoleEnabled(props)) {
+      LoggerContext ctx = getRootContext();
+      ctx.getLogger(ROOT_LOGGER_NAME).addAppender(newConsoleAppender(ctx, "root_console", logPattern));
+    }
+  }
+
+  /**
+   * Finds out whether we are in testing environment (usually ITs) and logs of all processes must be forward to
+   * App's System.out. This is specified by the value of property {@link #ALL_LOGS_TO_CONSOLE_PROPERTY}.
+   */
+  public boolean isAllLogsToConsoleEnabled(Props props) {
+    return props.valueAsBoolean(ALL_LOGS_TO_CONSOLE_PROPERTY, false);
+  }
+
+  @SafeVarargs
+  private static <T> Set<T> setOf(T... args) {
+    Set<T> res = new HashSet<>(args.length);
+    res.addAll(Arrays.asList(args));
+    return res;
+  }
+
+  /**
+   * Configure the log level of the root logger reading the value of property {@link #SONAR_LOG_LEVEL_PROPERTY}.
+   *
+   * @throws IllegalArgumentException if the value of {@link #SONAR_LOG_LEVEL_PROPERTY} is not one of {@link #ALLOWED_ROOT_LOG_LEVELS}
+   */
+  public Level configureRootLogLevel(Props props) {
+    return configureRootLogLevel(props, SONAR_LOG_LEVEL_PROPERTY);
+  }
+
+  /**
+   * Configure the log level of the root logger reading the value of specified property.
+   *
+   * @throws IllegalArgumentException if the value of the specified property is not one of {@link #ALLOWED_ROOT_LOG_LEVELS}
+   */
+  public Level configureRootLogLevel(Props props, String propertyKey) {
+    Level newLevel = Level.toLevel(props.value(propertyKey, Level.INFO.toString()), Level.INFO);
+    return configureRootLogLevel(newLevel);
+  }
+
+  /**
+   * Configure the log level of the root logger to the specified level.
+   *
+   * @throws IllegalArgumentException if the specified level is not one of {@link #ALLOWED_ROOT_LOG_LEVELS}
+   */
+  public Level configureRootLogLevel(Level newLevel) {
+    Logger rootLogger = getRootContext().getLogger(ROOT_LOGGER_NAME);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)